# Guided Capstone Step 4. Pre-Processing and Training Data Development

**The Data Science Method**  


1.   Problem Identification 


2.   Data Wrangling 
  
 
3.   Exploratory Data Analysis   

4.   **Pre-processing and Training Data Development**  
 * Create dummy or indicator features for categorical variables
  * Standardize the magnitude of numeric features
  * Split into testing and training datasets
  * Apply scaler to the testing set
5.   Modeling 
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

**<font color='teal'> Start by loading the necessary packages as we did in step 3 and printing out our current working directory just to confirm we are in the correct project directory. </font>**

In [3]:
import numpy as np
import pandas as pd
import os
import matplotlib.pyplot as plt
import seaborn as sns
import datetime
import scipy.stats
from statsmodels.formula import api as smf
%matplotlib inline

In [4]:
pwd

'D:\\Personal\\SpringBoard\\Full Course\\Week 4\\1588204778_GuidedCapstoneStep4_and_step5HL.ipynb'

**<font color='teal'>  Load the csv file you created in step 3, remember it should be saved inside your data subfolder and print the first five rows.</font>**

In [5]:
pd.read_csv('step3_output.csv').head()

Unnamed: 0                    Name       state  summit_elev  vertical_drop  \
0           1     Eaglecrest Ski Area      Alaska         2600           1540   
1           2        Hilltop Ski Area      Alaska         2090            294   
2           4     Sunrise Park Resort     Arizona        11100           1800   
3           8  Boreal Mountain Resort  California         7700            500   
4          10        Donner Ski Ranch  California         8012            750   

   trams  fastEight  fastSixes  fastQuads  quad  ...  SkiableTerrain_ac  \
0      0        0.0          0          0     0  ...              640.0   
1      0        0.0          0          0     0  ...               30.0   
2      0        0.0          0          1     2  ...              800.0   
3      0        0.0          0          1     1  ...              380.0   
4      0        0.0          0          0     0  ...              505.0   

   Snow Making_ac  daysOpenLastYear  yearsOpen  averageSnowfall  AdultWeekday  \
0            60.0              45.0       44.0            350.0          47.0   
1            30.0             150.0       36.0             69.0          30.0   
2            80.0             115.0       49.0            250.0          74.0   
3           200.0             150.0       54.0            400.0          49.0   
4            60.0             163.0       82.0            400.0          75.0   

   AdultWeekend  projectedDaysOpen  NightSkiing_ac  clusters  
0      53.00000               90.0             0.0         2  
1      34.00000              152.0            30.0         0  
2      78.00000              104.0            80.0         1  
3      64.16681              150.0           200.0         1  
4      75.00000              170.0             0.0         1  

[5 rows x 27 columns]

In [6]:
df = pd.read_csv('step3_output.csv')

In [7]:
df.head()

Unnamed: 0                    Name       state  summit_elev  vertical_drop  \
0           1     Eaglecrest Ski Area      Alaska         2600           1540   
1           2        Hilltop Ski Area      Alaska         2090            294   
2           4     Sunrise Park Resort     Arizona        11100           1800   
3           8  Boreal Mountain Resort  California         7700            500   
4          10        Donner Ski Ranch  California         8012            750   

   trams  fastEight  fastSixes  fastQuads  quad  ...  SkiableTerrain_ac  \
0      0        0.0          0          0     0  ...              640.0   
1      0        0.0          0          0     0  ...               30.0   
2      0        0.0          0          1     2  ...              800.0   
3      0        0.0          0          1     1  ...              380.0   
4      0        0.0          0          0     0  ...              505.0   

   Snow Making_ac  daysOpenLastYear  yearsOpen  averageSnowfall  AdultWeekday  \
0            60.0              45.0       44.0            350.0          47.0   
1            30.0             150.0       36.0             69.0          30.0   
2            80.0             115.0       49.0            250.0          74.0   
3           200.0             150.0       54.0            400.0          49.0   
4            60.0             163.0       82.0            400.0          75.0   

   AdultWeekend  projectedDaysOpen  NightSkiing_ac  clusters  
0      53.00000               90.0             0.0         2  
1      34.00000              152.0            30.0         0  
2      78.00000              104.0            80.0         1  
3      64.16681              150.0           200.0         1  
4      75.00000              170.0             0.0         1  

[5 rows x 27 columns]

## Create dummy features for categorical variables

**<font color='teal'> Create dummy variables for `state`. Add the dummies back to the dataframe and remove the original column for `state`. </font>**

Hint: you can see an example of how to execute this in Aiden's article on preprocessing [here](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-pre-processing-and-training-data-development-fd2d75182967). 

In [14]:
df.dtypes

Unnamed: 0             int64
Name                  object
state                 object
summit_elev            int64
vertical_drop          int64
trams                  int64
fastEight            float64
fastSixes              int64
fastQuads              int64
quad                   int64
triple                 int64
double                 int64
surface                int64
total_chairs           int64
Runs                 float64
TerrainParks         float64
LongestRun_mi        float64
SkiableTerrain_ac    float64
Snow Making_ac       float64
daysOpenLastYear     float64
yearsOpen            float64
averageSnowfall      float64
AdultWeekday         float64
AdultWeekend         float64
projectedDaysOpen    float64
NightSkiing_ac       float64
clusters               int64
dtype: object

In [18]:
dfo=df.select_dtypes(include=['object'])

In [19]:
dfo.head()

Name       state
0     Eaglecrest Ski Area      Alaska
1        Hilltop Ski Area      Alaska
2     Sunrise Park Resort     Arizona
3  Boreal Mountain Resort  California
4        Donner Ski Ranch  California

In [24]:
df.state.head()

0        Alaska
1        Alaska
2       Arizona
3    California
4    California
Name: state, dtype: object

In [26]:
df=pd.concat([df.drop('state', axis=1),pd.get_dummies(dfo.state, dummy_na=True)], axis=1)

In [27]:
df.head()

Unnamed: 0                    Name  summit_elev  vertical_drop  trams  \
0           1     Eaglecrest Ski Area         2600           1540      0   
1           2        Hilltop Ski Area         2090            294      0   
2           4     Sunrise Park Resort        11100           1800      0   
3           8  Boreal Mountain Resort         7700            500      0   
4          10        Donner Ski Ranch         8012            750      0   

   fastEight  fastSixes  fastQuads  quad  triple  ...  Pennsylvania  \
0        0.0          0          0     0       0  ...             0   
1        0.0          0          0     0       1  ...             0   
2        0.0          0          1     2       3  ...             0   
3        0.0          0          1     1       3  ...             0   
4        0.0          0          0     0       1  ...             0   

   Rhode Island  South Dakota  Utah  Vermont  Washington  West Virginia  \
0             0             0     0        0           0              0   
1             0             0     0        0           0              0   
2             0             0     0        0           0              0   
3             0             0     0        0           0              0   
4             0             0     0        0           0              0   

   Wisconsin  Wyoming  NaN  
0          0        0    0  
1          0        0    0  
2          0        0    0  
3          0        0    0  
4          0        0    0  

[5 rows x 57 columns]

In [28]:
df.dtypes

Unnamed: 0             int64
Name                  object
summit_elev            int64
vertical_drop          int64
trams                  int64
fastEight            float64
fastSixes              int64
fastQuads              int64
quad                   int64
triple                 int64
double                 int64
surface                int64
total_chairs           int64
Runs                 float64
TerrainParks         float64
LongestRun_mi        float64
SkiableTerrain_ac    float64
Snow Making_ac       float64
daysOpenLastYear     float64
yearsOpen            float64
averageSnowfall      float64
AdultWeekday         float64
AdultWeekend         float64
projectedDaysOpen    float64
NightSkiing_ac       float64
clusters               int64
Alaska                 uint8
Arizona                uint8
California             uint8
Colorado               uint8
Connecticut            uint8
Idaho                  uint8
Illinois               uint8
Indiana                uint8
Iowa          

## Standardize the magnitude of numeric features

**<font color='teal'> Using sklearn preprocessing standardize the scale of the features of the dataframe except the name of the resort which we done't need in the dataframe for modeling, so it can be droppped here as well. Also, we want to hold out our response variable(s) so we can have their true values available for model performance review. Let's set `AdultWeekend` to the y variable as our response for scaling and modeling. Later we will go back and consider the `AdultWeekday`, `dayOpenLastYear`, and `projectedDaysOpen`. For now leave them in the development dataframe. </font>**

In [32]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X = df.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y = df.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler = preprocessing.StandardScaler().fit(X)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled=scaler.transform(X) 

## Split into training and testing datasets

**<font color='teal'> Using sklearn model selection import train_test_split, and create a 75/25 split with the y = `AdultWeekend`. We will start by using the adult weekend ticket price as our response variable for modeling.</font>**

In [33]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y = y.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.25, random_state=1)

Here we start the actual modeling work. First let's fit a multiple linear regression model to predict the `AdultWeekend` price.

# Guided Capstone Step 5. Modeling


This is the fifth step in the Data Science Method. In the previous steps you cleaned and prepared the datasets. Now it's time to get into the most exciting part: modeling! In this exercise, you'll build three different models and compare each model's performance. In the end, you'll choose the best model for demonstrating insights to Big Mountain management.



### **The Data Science Method**  


1.   Problem Identification 

2.   Data Wrangling 
  
3.   Exploratory Data Analysis 
 
4.   Pre-processing and Training Data Development

5.   **Modeling**
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

## Fit Models with a Training Dataset

**<font color='teal'> Using sklearn, fit the model on your training dataset.</font>**

#### Model 1

In [34]:
#all first model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm = linear_model.LinearRegression()
model = lm.fit(X_train,y_train)

**<font color='teal'> Predict on the testing dataset and score the model performance with the y_test set and the y-pred values. The explained variance is a measure of the variation explained by the model. This is also known as the R-squared value. </font>**

Hint: you will have to use the `predict()` method here as it's used in this [DSM article](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-modeling-56b4233cad1b) about modeling. 

In [35]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred = model.predict(X_test)

## Review Model Outcomes — Iterate over additional models as needed

In [0]:
# You might want to use the explained_variance_score() and mean_absolute_error() metrics.
# To do so, you will need to import them from sklearn.metrics. 
# You can plug y_test and y_pred into the functions to evaluate the model

In [49]:
# rmae > root mean absolute error | rsv > R-Squared value
from math import sqrt
rmae = sqrt(mean_absolute_error(y_test,y_pred))
rsv = explained_variance_score(y_test,y_pred)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 11817727.612711959 & rsv: -8.225984189749318e+26


**<font color='teal'> Print the intercept value from the linear model. </font>**

Hint: our linear regression model `lm` has an attribute `intercept_` for the intercept

In [50]:
print(lm.intercept_)

-35912233524348.12


**<font color='teal'> The intercept is the mean `AdultWeekend` price for all the resorts given the other characteristics. The addition or subtraction of each of the coefficient values in the regression are numeric adjustments applied to the intercept to provide a particular observation's value for the resulting `AdultWeekend` value. Also, because we took the time to scale our x values in the training data, we can compare each of the coeeficients for the features to determine the feature importances. Print the coefficient values from the linear model and sort in descending order to identify the top ten most important features.</font>** 


Hint: make sure to review the absolute value of the coefficients, because the adjustment may be positive or negative, but what we are looking for is the magnitude of impact on our response variable.

In [59]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
pd.DataFrame(abs(lm.coef_), X.columns, columns=['Coefficient']).sort_values('Coefficient', ascending=False)


Coefficient
fastEight          7.533069e+14
fastSixes          4.043310e+14
total_chairs       2.158517e+14
South Dakota       1.281793e+14
Washington         1.281793e+14
Rhode Island       1.281793e+14
double             1.235164e+14
trams              1.213723e+14
surface            1.139628e+14
triple             1.047539e+14
quad               6.019971e+13
fastQuads          5.245467e+13
Michigan           1.843989e+13
Wisconsin          1.843989e+13
New York           1.758545e+13
New Hampshire      1.666585e+13
Pennsylvania       1.666585e+13
California         1.458109e+13
Minnesota          1.337771e+13
Connecticut        1.202506e+13
Maine              1.202506e+13
New Mexico         1.202506e+13
Vermont            1.202506e+13
Massachusetts      1.046544e+13
Ohio               1.046544e+13
Wyoming            1.046544e+13
Alaska             8.586778e+12
Illinois           8.586778e+12
Indiana            8.586778e+12
Idaho              8.586778e+12
Colorado           8.586778e+12
North Carolina     8.586778e+12
Utah               8.586778e+12
West Virginia      8.586778e+12
Iowa               6.101172e+12
Maryland           6.101172e+12
Arizona            6.101172e+12
Missouri           6.101172e+12
Nevada             6.101172e+12
Unnamed: 0         2.168661e+01
summit_elev        1.113153e+01
AdultWeekday       8.812500e+00
clusters           4.656250e+00
daysOpenLastYear   3.000000e+00
projectedDaysOpen  1.968750e+00
LongestRun_mi      1.296875e+00
TerrainParks       1.171875e+00
SkiableTerrain_ac  9.687500e-01
vertical_drop      6.992496e-01
Snow Making_ac     5.625000e-01
Runs               5.390625e-01
NightSkiing_ac     4.375000e-01
averageSnowfall    2.753906e-01
yearsOpen          3.125000e-02
NaN                0.000000e+00

**<font color='teal'>You should see that the top ten important features are different states. However, the state is not something the managers at the Big Mountain Resort can do anything about. Given that we care more about actionable traits associated with ticket pricing, rebuild the model without the state features and compare the results. </font>**

Hint: Try to construct another model using exactly the steps we followed above. 

#### Model 2

In [78]:
df.columns

Index([       'Unnamed: 0',              'Name',       'summit_elev',
           'vertical_drop',             'trams',         'fastEight',
               'fastSixes',         'fastQuads',              'quad',
                  'triple',            'double',           'surface',
            'total_chairs',              'Runs',      'TerrainParks',
           'LongestRun_mi', 'SkiableTerrain_ac',    'Snow Making_ac',
        'daysOpenLastYear',         'yearsOpen',   'averageSnowfall',
            'AdultWeekday',      'AdultWeekend', 'projectedDaysOpen',
          'NightSkiing_ac',          'clusters',            'Alaska',
                 'Arizona',        'California',          'Colorado',
             'Connecticut',             'Idaho',          'Illinois',
                 'Indiana',              'Iowa',             'Maine',
                'Maryland',     'Massachusetts',          'Michigan',
               'Minnesota',          'Missouri',            'Nevada',
           'New Hamp

In [61]:
df1=df.select_dtypes(exclude=['uint8'])

In [77]:
df1.columns

Index(['Unnamed: 0', 'Name', 'summit_elev', 'vertical_drop', 'trams',
       'fastEight', 'fastSixes', 'fastQuads', 'quad', 'triple', 'double',
       'surface', 'total_chairs', 'Runs', 'TerrainParks', 'LongestRun_mi',
       'SkiableTerrain_ac', 'Snow Making_ac', 'daysOpenLastYear', 'yearsOpen',
       'averageSnowfall', 'AdultWeekday', 'AdultWeekend', 'projectedDaysOpen',
       'NightSkiing_ac', 'clusters'],
      dtype='object')

In [70]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X1 = df1.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y1 = df1.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler1 = preprocessing.StandardScaler().fit(X1)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled1=scaler1.transform(X1) 

In [71]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y1 = y1.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train1, X_test1, y_train1, y_test1 = train_test_split(X_scaled1, y1, test_size=0.25, random_state=1)

In [72]:
#all second model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm1 = linear_model.LinearRegression()
model1 = lm1.fit(X_train1,y_train1)

In [73]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred1 = model1.predict(X_test1)

In [74]:
# rmae > root mean absolute error | rsv > R-Squared value
from math import sqrt
rmae = sqrt(mean_absolute_error(y_test1,y_pred1))
rsv = explained_variance_score(y_test1,y_pred1)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 2.2788074877610196 & rsv: 0.7883386063227413


In [75]:
print(lm1.intercept_)

59.436585781076644


In [76]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
pd.DataFrame(abs(lm1.coef_), X1.columns, columns=['Coefficient']).sort_values('Coefficient', ascending=False)

Coefficient
AdultWeekday       8.956771e+00
clusters           3.553234e+00
daysOpenLastYear   3.370794e+00
projectedDaysOpen  2.228691e+00
SkiableTerrain_ac  2.158073e+00
summit_elev        1.858674e+00
LongestRun_mi      1.719355e+00
Runs               1.575613e+00
surface            1.206041e+00
triple             1.172055e+00
double             9.429151e-01
averageSnowfall    7.731362e-01
TerrainParks       7.134972e-01
total_chairs       6.382109e-01
fastQuads          5.373438e-01
Unnamed: 0         5.065743e-01
NightSkiing_ac     4.175915e-01
vertical_drop      4.114814e-01
Snow Making_ac     3.679563e-01
yearsOpen          1.765732e-01
quad               1.291308e-01
trams              1.332268e-15
fastSixes          0.000000e+00
fastEight          0.000000e+00

**<font color='teal'> When reviewing our new model coefficients, we see `summit_elev` is now in the number two spot. This is also difficult to change from a management prespective and highly correlated with `base_elev` and `vertical_drop`.  This time, rebuild the model without the state features and without the `summit_elev` and without `base_elev`and compare the results. </font>**

#### Model 3

In [79]:
df2=df1.drop(columns='summit_elev')

In [80]:
df2.head()

Unnamed: 0                    Name  vertical_drop  trams  fastEight  \
0           1     Eaglecrest Ski Area           1540      0        0.0   
1           2        Hilltop Ski Area            294      0        0.0   
2           4     Sunrise Park Resort           1800      0        0.0   
3           8  Boreal Mountain Resort            500      0        0.0   
4          10        Donner Ski Ranch            750      0        0.0   

   fastSixes  fastQuads  quad  triple  double  ...  SkiableTerrain_ac  \
0          0          0     0       0       4  ...              640.0   
1          0          0     0       1       0  ...               30.0   
2          0          1     2       3       1  ...              800.0   
3          0          1     1       3       1  ...              380.0   
4          0          0     0       1       5  ...              505.0   

   Snow Making_ac  daysOpenLastYear  yearsOpen  averageSnowfall  AdultWeekday  \
0            60.0              45.0       44.0            350.0          47.0   
1            30.0             150.0       36.0             69.0          30.0   
2            80.0             115.0       49.0            250.0          74.0   
3           200.0             150.0       54.0            400.0          49.0   
4            60.0             163.0       82.0            400.0          75.0   

   AdultWeekend  projectedDaysOpen  NightSkiing_ac  clusters  
0      53.00000               90.0             0.0         2  
1      34.00000              152.0            30.0         0  
2      78.00000              104.0            80.0         1  
3      64.16681              150.0           200.0         1  
4      75.00000              170.0             0.0         1  

[5 rows x 25 columns]

In [81]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X2 = df2.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y2 = df2.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler2 = preprocessing.StandardScaler().fit(X2)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled2=scaler2.transform(X2) 

In [82]:
# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y2 = y2.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train2, X_test2, y_train2, y_test2 = train_test_split(X_scaled2, y2, test_size=0.25, random_state=1)

In [83]:
#all second model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm2 = linear_model.LinearRegression()
model2 = lm2.fit(X_train2,y_train2)

In [84]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y_pred2 = model2.predict(X_test2)

In [85]:
# rmae > root mean absolute error | rsv > R-Squared value
from math import sqrt
rmae = sqrt(mean_absolute_error(y_test2,y_pred2))
rsv = explained_variance_score(y_test2,y_pred2)
print(f"rmae: {rmae} & rsv: {rsv}")

rmae: 2.363284082060734 & rsv: 0.7531358868579863


In [86]:
print(lm2.intercept_)

59.455907184424454


In [87]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
pd.DataFrame(abs(lm2.coef_), X2.columns, columns=['Coefficient']).sort_values('Coefficient', ascending=False)

Coefficient
AdultWeekday       8.928189e+00
clusters           3.228784e+00
daysOpenLastYear   3.210167e+00
SkiableTerrain_ac  2.774243e+00
projectedDaysOpen  2.501706e+00
LongestRun_mi      2.182586e+00
Runs               1.560558e+00
averageSnowfall    1.288289e+00
double             1.060907e+00
triple             8.999866e-01
surface            8.956316e-01
Snow Making_ac     8.737414e-01
TerrainParks       8.701815e-01
Unnamed: 0         6.623852e-01
total_chairs       6.169381e-01
NightSkiing_ac     2.353834e-01
fastQuads          2.306837e-01
vertical_drop      1.511780e-01
quad               3.623161e-02
yearsOpen          2.613249e-02
fastSixes          3.552714e-15
trams              8.881784e-16
fastEight          0.000000e+00

## Identify the Final Model

**<font color='teal'> Review the model performances in the table below and choose the best model for proving insights to Big Mountain management about what features are driving ski resort lift ticket prices. Type your choice in the final markdown cell — you will discuss this selection more in the next step of the guided casptone. </font>**

| Model | Explained Variance| Mean Absolute Error|Features Dropped|
| --- | --- | --- | --- |
| Model 1. | -8.225984189749318e+26 | 11817727.612711959 |-|
| Model 2. | 0.7883386063227413 | 2.2788074877610196 |'state'|
| Model 3. | 0.7531358868579863 | 2.363284082060734 |'state','summit_elev','base_elev'|

Model Selection:

# Model 3